lib/saluki-env/src/autodiscovery/providers/local.rs

```diff
@@ -66,32 +76,76 @@ impl LocalAutoDiscoveryProvider {
     }
 }
 
+#[derive(Debug, Deserialize)]
+struct CheckConfig {
+    init_config: serde_yaml::Mapping,
+    instances: Vec<serde_yaml::Mapping>,
+}
+
 /// Parse a YAML file into a Config object
 async fn parse_config_file(path: &PathBuf) -> Result<(String, Config), GenericError> {
     let content = fs::read_to_string(path).await?;
 
+    let check_config: CheckConfig = match serde_yaml::from_str(&content) {
+        Ok(read) => read,
+        Err(e) => {
+            warn!("Can't decode yaml as check configuration: {}", content);
```

Comment:
You're already emitting a warning log when you call `parse_config_file` and it returns `Err(e)`, so no need to also emit a warning log here too.

Reply:
removed 🔥